src/NUnitFramework/framework/Internal/Execution/SimpleWorkItem.cs

```diff
@@ -153,6 +154,14 @@ private TestCommand MakeTestCommand()
                     command = new TimeoutCommand(command, timeout);
 #endif
 
+                // warn if we have more than one IRepeatTest (Retry / Repeat) Attribute
+                if (method.GetAttributes<IRepeatTest>(true).Length > 1)
+                    return new WarnCommand(_testMethod, "Test case has both RetryAttribute and RepeatAttribute, which could give unpredictable results.");
```

Comment:
This puts a small bit more information into workitem than we would like it to have. Can we just say that multiple attributes that repeat a test may cause problems? Remember, a user attribute could implement the interface.